File: CS/WizardConnectionStorageService/Form1.cs
using DevExpress.DashboardWin;
using DevExpress.DataAccess;
using DevExpress.DataAccess.ConnectionParameters;
using DevExpress.DataAccess.Sql;
using DevExpress.DataAccess.Wizard.Model;
using DevExpress.DataAccess.Wizard.Services;
using DevExpress.XtraBars.Ribbon;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace WizardCustomizationExample1
{
    public partial class Form1 : RibbonForm
    {
        public Form1()
        {
            InitializeComponent();

            dashboardDesigner1.ServiceContainer.RemoveService(typeof(IConnectionStorageService));
            dashboardDesigner1.ServiceContainer.AddService(typeof(IConnectionStorageService), new CustomConnectionStorageService(dashboardDesigner1));

            MsSqlConnectionParameters cpNorthwind = new MsSqlConnectionParameters()
            {
                ServerName = "localhost",
                AuthorizationType = MsSqlAuthorizationType.Windows,
                DatabaseName = "Northwind"
            };
            MsSqlConnectionParameters cpAdventureWorks = new MsSqlConnectionParameters()
            {
                ServerName = "localhost",
                AuthorizationType = MsSqlAuthorizationType.Windows,
                DatabaseName = "AdventureWorksDW2012"
            };
            MsSqlConnectionParameters cpContosoRetail = new MsSqlConnectionParameters()
            {
                ServerName = "localhost",
                AuthorizationType = MsSqlAuthorizationType.Windows,
                DatabaseName = "ContosoRetailDW"
            };
            dashboardDesigner1.CustomDataConnections.Add(new DataConnection("Local Northwind Connection", cpNorthwind));
            dashboardDesigner1.CustomDataConnections.Add(new DataConnection("Local AdventureWorks Connection", cpAdventureWorks));
            dashboardDesigner1.CustomDataConnections.Add(new DataConnection("Local ContosoRetail Connection", cpContosoRetail));
        }
    }
    public class CustomConnectionStorageService : IConnectionStorageService
    {
        readonly DashboardDesigner designer;

        public CustomConnectionStorageService(DashboardDesigner designer)
        {
            this.designer = designer;
        }
        public bool CanSaveConnection
        {
            get { return false; }
        }
        public bool Contains(string connectionName)
        {
            return designer.CustomDataConnections.Any(connection => connection.Name == connectionName);
        }

        public IEnumerable<SqlDataConnection> GetConnections()
        {
            return designer.CustomDataConnections;
        }

        public void SaveConnection(string connectionName, IDataConnection connection, bool saveCredentials)
        {
            throw new NotSupportedException();
        }
    }
}

File: VB/WizardConnectionStorageService/Form1.vb
Imports DevExpress.DashboardWin
Imports DevExpress.DataAccess
Imports DevExpress.DataAccess.ConnectionParameters
Imports DevExpress.DataAccess.Sql
Imports DevExpress.DataAccess.Wizard.Model
Imports DevExpress.DataAccess.Wizard.Services
Imports DevExpress.XtraBars.Ribbon
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.ComponentModel.Design
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace WizardCustomizationExample1
    Partial Public Class Form1
        Inherits RibbonForm

        Public Sub New()
            InitializeComponent()

            dashboardDesigner1.ServiceContainer.RemoveService(GetType(IConnectionStorageService))
            dashboardDesigner1.ServiceContainer.AddService(GetType(IConnectionStorageService), New CustomConnectionStorageService(dashboardDesigner1))

            Dim cpNorthwind As New MsSqlConnectionParameters() With {.ServerName = "localhost", .AuthorizationType = MsSqlAuthorizationType.Windows, .DatabaseName = "Northwind"}
            Dim cpAdventureWorks As New MsSqlConnectionParameters() With {.ServerName = "localhost", .AuthorizationType = MsSqlAuthorizationType.Windows, .DatabaseName = "AdventureWorksDW2012"}
            Dim cpContosoRetail As New MsSqlConnectionParameters() With {.ServerName = "localhost", .AuthorizationType = MsSqlAuthorizationType.Windows, .DatabaseName = "ContosoRetailDW"}
            dashboardDesigner1.CustomDataConnections.Add(New DataConnection("Local Northwind Connection", cpNorthwind))
            dashboardDesigner1.CustomDataConnections.Add(New DataConnection("Local AdventureWorks Connection", cpAdventureWorks))
            dashboardDesigner1.CustomDataConnections.Add(New DataConnection("Local ContosoRetail Connection", cpContosoRetail))
        End Sub
    End Class
    Public Class CustomConnectionStorageService
        Implements IConnectionStorageService

        Private ReadOnly designer As DashboardDesigner

        Public Sub New(ByVal designer As DashboardDesigner)
            Me.designer = designer
        End Sub
        Public ReadOnly Property CanSaveConnection() As Boolean Implements IConnectionStorageService.CanSaveConnection
            Get
                Return False
            End Get
        End Property
        Public Function Contains(ByVal connectionName As String) As Boolean Implements IConnectionStorageService.Contains
            Return designer.CustomDataConnections.Any(Function(connection) connection.Name = connectionName)
        End Function

        Public Function GetConnections() As IEnumerable(Of SqlDataConnection) Implements IConnectionStorageService.GetConnections
            Return designer.CustomDataConnections
        End Function

        Public Sub SaveConnection(ByVal connectionName As String, ByVal connection As IDataConnection, ByVal saveCredentials As Boolean) Implements IConnectionStorageService.SaveConnection
            Throw New NotSupportedException()
        End Sub
    End Class
End Namespace
